datafusion/expr/src/expr_schema.rs

```diff
@@ -314,11 +322,17 @@ impl ExprSchemable for Expr {
                 }
             }
             Expr::Cast(Cast { expr, .. }) => expr.nullable(input_schema),
+            Expr::AggregateFunction(AggregateFunction { func_def, .. }) => {
+                match func_def {
+                    AggregateFunctionDefinition::BuiltIn(fun) => fun.nullable(),
+                    // TODO: UDF should be able to customize nullability
```

Comment:
I'm not sure about that we need to customize nullability, can we get the nullabiity based on expression in the function?

Reply:
Maybe we just need to add `field` to UDAFImpl, so they can define their own field

Reply:
> I'm not sure about whether we need to customize nullability, can we get the nullabiity based on expression in the function?

I think probably ArrayAgg is the function that we need the nullability be `false` 🤔

Reply:
> I'm not sure about whether we need to customize nullability, can we get the nullabiity based on expression in the function?

But that will need to get it from the specific UDAF implementation right?  

> If we need it, we can add field to UDAFImpl, so they can define their own field. state_field is already customizable

That would resolve it.

> I think probably ArrayAgg is the function that we need the nullability be `false` 🤔

There is also `count` and the the count like e.g approx_distinct etc.